File: examples/python/RVC4/ToF/tof_params_minimal.py
#!/usr/bin/env python3
"""Minimal RVC4 ToF example with explicit parameter setup."""

import depthai as dai

# --- build parameters ---
BOARD_SOCKET = dai.CameraBoardSocket.AUTO
FPS = 10.0
PRESET = dai.ToFPreset.MID_RANGE         # IPP profile: LOW_RANGE / MID_RANGE / HIGH_RANGE / FAST_OBJECTS

with dai.Pipeline() as pipeline:
    tof = pipeline.create(dai.node.ToF)

    # build resolves socket; preset= selects the IPP profile.
    # VD55H1 capture mode is fixed to F3_FULL internally (not user-selectable).
    tof.build(
        BOARD_SOCKET,
        fps=FPS,
        preset=PRESET,
    )

    # tune initialConfig after build() so values are not overwritten
    cfg = tof.initialConfig
    cfg.enableBilateralFilter = True
    cfg.enableTemporalNoiseReduction = True
    cfg.enableFlyingPixelCorrection = True
    cfg.enableRadialToPerp = False
    cfg.phaseUnwrapErrorThreshold = 75

    depth_q = tof.depth.createOutputQueue()
    cfg_q = tof.inputConfig.createInputQueue()

    pipeline.start()
    print(
        f"socket={tof.getBoardSocket()} output={tof.getOutputResolution()} raw={tof.getRawResolution()} "
        f"auto_cam={tof.getCamera() is not None}"
    )

    depth = depth_q.get()
    print(f"depth frame: {depth.getWidth()}x{depth.getHeight()}")

    # optional: push updated IPP params at runtime
    cfg.enableRadialToPerp = True
    cfg_q.send(cfg)

    depth2 = depth_q.get()
    print(f"depth frame after runtime config: {depth2.getWidth()}x{depth2.getHeight()}")

File: examples/cpp/RVC4/ToF/tof_params_minimal.cpp
/**
 * RVC4 ToF example: build(), initialConfig tuning, and runtime inputConfig updates.
 */

#include <iostream>
#include <memory>

#include "depthai/common/ToFPreset.hpp"
#include "depthai/depthai.hpp"
#include "depthai/pipeline/datatype/ToFConfig.hpp"
#include "depthai/pipeline/node/ToF.hpp"

int main() {
    dai::Pipeline pipeline;

    auto tof = pipeline.create<dai::node::ToF>();

    dai::ToFBuildOptions options;
    options.boardSocket = dai::CameraBoardSocket::AUTO;
    options.fps = 10.f;  // VD55H1 capture mode fixed to F3_FULL internally
    options.preset = dai::ToFPreset::MID_RANGE;
    tof->build(options);

    // Tune IPP fields after build() so build(preset=...) does not overwrite them.
    auto& cfg = *tof->initialConfig;
    cfg.enableBilateralFilter = true;
    cfg.enableTemporalNoiseReduction = true;
    cfg.enableFlyingPixelCorrection = true;
    cfg.enableRadialToPerp = false;
    cfg.phaseUnwrapErrorThreshold = 75;

    const auto depthQueue = tof->depth.createOutputQueue();
    const auto cfgQueue = tof->inputConfig.createInputQueue();

    pipeline.start();

    const auto [outW, outH] = tof->getOutputResolution();
    const auto [rawW, rawH] = tof->getRawResolution();
    std::cout << "socket=" << static_cast<int>(tof->getBoardSocket()) << " output=" << outW << "x" << outH << " raw=" << rawW << "x" << rawH
              << " auto_cam=" << (tof->getCamera() != nullptr) << std::endl;

    if(const auto depth = depthQueue->get<dai::ImgFrame>()) {
        std::cout << "depth frame: " << depth->getWidth() << "x" << depth->getHeight() << std::endl;
    }

    cfg.enableRadialToPerp = true;
    cfgQueue->send(std::make_shared<dai::ToFConfig>(cfg));

    if(const auto depth2 = depthQueue->get<dai::ImgFrame>()) {
        std::cout << "depth frame after runtime config: " << depth2->getWidth() << "x" << depth2->getHeight() << std::endl;
    }

    return 0;
}
